craft_providers/lxd/launcher.py

```diff
@@ -486,16 +485,35 @@ def _set_id_map(
     :param remote: LXD remote to create instance on.
     :param uid: The uid to be mapped. If not supplied, the current user's uid is used.
     """
+    configured_id_map = ""
     if uid is None:
         uid = os.getuid()
 
+    # Retry for idmap to be set
     lxc.config_set(
         instance_name=instance.instance_name,
         key="raw.idmap",
         value=f"both {uid!s} 0",
         project=project,
         remote=remote,
     )
+    configured_id_map = lxc.config_get(
```

Comment:
I wonder if we should add a comment explaining that we do `config_set()` followed by `config_get()` to catch the bug in https://github.com/canonical/lxd/issues/12189 /  https://github.com/canonical/lxd/pull/12191.

Reply:
I can add some comments on that